cloudferrylib/os/actions/is_boot_image_deleted.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+import copy
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import utils as utl
+
+PATH_RECREATE_IMAGE = 0
+DEFAULT = 1
+
+
+class IsBootImageDeleted(action.Action):
+    def run(self, info=None, missing_images={}, **kwargs):
+        self.set_next_path(DEFAULT)
+        if missing_images:
+            self.set_next_path(PATH_RECREATE_IMAGE)
+        return {
```

Comment:
Since it doesn't update existing dictionary, let's just remove this `return` statement: less code is better code.